sktime/transformations/base.py

```diff
@@ -671,10 +673,11 @@ def _vectorized_transform(self, X, X_input_mtype=None, y=None, inverse=False):
         """Vectorized application of transform or inverse, and convert back."""
         if X_input_mtype is None:
             X_input_mtype = mtype(X, as_scitype=["Series", "Panel"])
-        if y is not None:
-            ValueError(
-                "no default behaviour if _fit does not support Panel, "
-                " but X is Panel and y is not None"
+        if y is not None and self.get_tag("y_inner_mtype") != "None":
+            raise ValueError(
+                "no default behaviour if _fit does not support Panel and "
+                'self.get_tag("y_inner_mtype") is not "None",'
```

Comment:
If this is intended as a message to the user it might not be as helpful as anticipated. Consider adding a suggestion about what to change. BTW this is repeated 3rd time at this point, maybe it could be a custom Exception?